### PR TITLE
moving training pipelines from cuda 11.5 to 11.6 and deprecating 11.3 (packaging pipeline)

### PR DESCRIPTION
**Pull request by AdamLouly**
**Description**: 
Moving training pipelines from cuda 11.5 to 11.6.
**Motivation and Context**
- Updating cuda to 11.6 for training pipelines.
- Deprecating cuda 11.3 pipelines.

### CODE REVIEW

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_3 [hunk 1]
```diff
@@ -1,187 +0,0 @@
-ARG BASEIMAGE=nvcr.io/nvidia/cuda:11.3.1-cudnn8-devel-centos7
```
Comment (baijumeswani) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_3:1]: This file is needed by our CI pipeline I believe. I think you can submit another PR for updating the CUDA version in our CI pipeline to 11.6.

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_6 [hunk 2]
```diff
@@ -1,4 +1,4 @@
-ARG BASEIMAGE=nvidia/cuda:11.5.0-cudnn8-devel-centos7
+ARG BASEIMAGE=nvidia/cuda:11.6.0-cudnn8-devel-centos7
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_6:1]: I suggest using the latest version: 11.6.2-cudnn8-devel-centos7

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_3 (hunk 1 above)
Comment (baijumeswani) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_3:1]: On reviewing again, this file is not needed as far as I can tell. It is ok to remove.

**Comment by AdamLouly** [comment]
Build Succeeded for the new pipeline : [See Logs Here ](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=729354&view=logs&j=8025e754-8287-57d5-c6ee-43803c3b3545)

**Review by snnn — APPROVED** [review]

**Review by baijumeswani — APPROVED** [review]

**Review by baijumeswani — APPROVED** [review]

**Review by snnn — APPROVED** [review]

**Comment by AdamLouly** [comment]
Thanks for the review!

**Pull request merged**